models/base.py

```diff
@@ -430,10 +443,16 @@ def get_name_in_current_ctx(self) -> str:
     cur_scope_abs = cur_scope.get_abs_name_ctx_list()
     self_name_abs = self.get_abs_name_ctx_list()
     assert cur_scope_abs[0] is self_name_abs[0]  # same root
-    assert len(cur_scope_abs) >= len(self_name_abs)  # not implemented otherwise
     common_len = 0
-    while cur_scope_abs[common_len + 1] is self_name_abs[common_len + 1]:
+    max_common_len = max(len(cur_scope_abs), len(cur_scope_abs))
+    while common_len+1 < max_common_len and cur_scope_abs[common_len+1] is self_name_abs[common_len+1]:
       common_len += 1
+    if common_len < len(self_name_abs) and len(cur_scope_abs) < len(self_name_abs):  # grabbing a sublayer
```

Comment:
This should not be a separate case. Then you would still not cover all cases (e.g. first going up and then going down). Just merge it with the code below. I think it would be very similar to the `relpath` code which I posted.